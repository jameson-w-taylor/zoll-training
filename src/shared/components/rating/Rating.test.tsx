import { render, screen, waitFor } from '@testing-library/react';
import { ionFireEvent as fireEvent } from '@ionic/react-test-utils';
import { Rating } from './Rating';

describe('<Rating />', () => {
  describe('when enabled', () => {
    let props: any;

    beforeEach(() => (props = { onRatingChange: jest.fn() }));

    it('renders consistently', () => {
      const { asFragment } = render(<Rating {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('sets the opacity to 1', () => {
      render(<Rating {...props} />);
      const rating = screen.getByTestId(/rating/) as HTMLDivElement;
      expect(rating.style.opacity).toEqual('1');
    });

    it('sets the rating on click', async () => {
      render(<Rating {...props} />);
      const fourStars = screen.getByTestId(/Rate 4 stars/);
      fireEvent.click(fourStars);
      await waitFor(() => expect(props.onRatingChange).toHaveBeenCalledTimes(1));
      expect(props.onRatingChange).toBeCalledWith(4);
    });
  });

  describe('when disabled', () => {
    let props: any;
    beforeEach(() => (props = { onRatingChange: jest.fn(), disabled: true }));

    it('renders consistently', () => {
      const { asFragment } = render(<Rating {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('sets the opacity to 0.25', () => {
      render(<Rating {...props} />);
      const rating = screen.getByTestId(/rating/) as HTMLDivElement;
      expect(rating.style.opacity).toEqual('0.25');
    });
  });

  afterEach(() => {});
});
