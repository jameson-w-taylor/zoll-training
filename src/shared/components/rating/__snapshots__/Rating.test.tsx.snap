// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> when disabled renders consistently 1`] = `
<DocumentFragment>
  <div
    class="rating"
    data-testid="rating"
    style="opacity: 0.25;"
  >
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
  </div>
</DocumentFragment>
`;

exports[`<Rating /> when enabled renders consistently 1`] = `
<DocumentFragment>
  <div
    class="rating"
    data-testid="rating"
    style="opacity: 1;"
  >
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
    Object {
      "hasAriaHidden": [Function],
      "iconName": null,
      "inheritedAttributes": Object {},
    }
  </div>
</DocumentFragment>
`;
